### ISSUE TITLE
View of list of variations in cart is different depending on how vars are set up

### ISSUE THREAD

**Issue by yukikatayama** [woocommerce/woocommerce#22128]
**Describe the bug**
Reported in 8569512-hc

When two attributes are setup to "Any Variables" then the variations shows in a list in cart, while when variations are created separately, they show on one line.

"Any" Set up:

![](http://cld.wthms.co/fYImqD+)
Link to screenshot: http://cld.wthms.co/fYImqD

And the separate set up:

![](http://cld.wthms.co/GT3MLO+)
Link to screenshot: http://cld.wthms.co/GT3MLO

Shows:

![](https://cld.wthms.co/n8Q3tW+)
Link to screenshot: https://cld.wthms.co/n8Q3tW

**Expected behavior**

That they would list variations one or the other (list for makes the most sense).

When testing with three attributes, both show correctly:

![](https://cld.wthms.co/fUN4zY+)
Link to screenshot: https://cld.wthms.co/fUN4zY

**WordPress Environment**
WC 3.5.2

**Comment by faisal-alvi** [comment]
Hi @yukikatayama , I have submitted a PR for this fix but it requires review. Please wait until its reviewed. Have a nice day :)

**Comment by mikejolley** [comment]
Hi @yukikatayama and @faisal-alvi 

When you give a variation attributes, we can generate a title for the variation matching and containing those attributes (create all var from attribute screenshot).

If a variation has 'any', we cannot generate a title beause we don't know which specific attribute is relevent.

If an attribute is not contained in the product/variation title, it goes in the list format (your expected behavior).

Having attributes in the title has benefits such as making it easier to search for specific variations in backend, and keeping the cart page simple/clean. This can be customised if not wanted using some custom code, @SiR-DanieL covered on his blog and there may be some other docs elsewhere I'm not aware of https://nicola.blog/2017/06/01/remove-attribute-from-product-title/

tl;dr this is intentional and is not a bug.

Thanks
